libctru/source/archive_dev.c

```diff
@@ -980,12 +980,23 @@ archive_rename(struct _reent *r,
   }
 
   rc = FSUSER_RenameFile(sourceDevice->archive, fs_path_old, sourceDevice->archive, fs_path_new);
-  if(R_SUCCEEDED(rc))
-    return 0;
+  /* if the file at the target destination exists, overwrite it */
+  if(rc == 0xC82044BE) {
```

Comment:
I don't like the magic numbers here. We have some macros that might be useful in https://github.com/devkitPro/libctru/blob/master/libctru/include/3ds/result.h

I think `if ( R_FAILED(rc) && R_DESCRIPTION(rc) == RD_ALREADY_EXISTS`) should probably do it.